### PR TITLE
Elevator System

### PR DESCRIPTION
**Pull request by Souheil-Yazji**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,4 +12,12 @@ Group 4
 - Souheil Yazji
 
 ## Contributions
-Use the following link for [Commit Convention](https://www.conventionalcommits.org/en/v1.0.0/). All contributions to the project should pretain to an [Issue](https://github.com/mpfan/SYSC-3303-Elevator-Simulator/issues). Individual contributions to the project should be made through Pull Requests into the Dev branch.
\ No newline at end of file
+Use the following link for [Commit Convention](https://www.conventionalcommits.org/en/v1.0.0/). All contributions to the project should pretain to an [Issue](https://github.com/mpfan/SYSC-3303-Elevator-Simulator/issues). Individual contributions to the project should be made through Pull Requests into the Dev branch.
```
Comment (dershao) [inline README.md:15]: if possible, can you make this separate from this branch?

File: src/uml2.ucls [hunk 2]
```diff
@@ -0,0 +1,254 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dershao) [inline src/uml2.ucls]: this as well

**Review by dershao — CHANGES_REQUESTED** [review]

**Comment by Souheil-Yazji** [comment]
@dershao  does your PR cover most of this anyways? If so, we can close this one and I'll make another PR with the readme and uml

**Comment by Souheil-Yazji** [comment]
test

**Comment by dershao** [comment]
My PR does not cover it. 

I guess we can keep it since I feel like the elevator system is close to be merged anyways.

**Review by dershao — APPROVED** [review]

File: src/elevator/ElevatorSystem.java [hunk 3]
```diff
@@ -1,58 +1,77 @@
...
+ */
 public class ElevatorSystem implements Runnable {
-	
-	private Requestable scheduler; 
-	private List<Elevator> elevators;
-	private Queue<Message> requests;
-	
+
+	private Scheduler scheduler;
+	private Elevator ele1;
```
Comment (mpfan) [inline src/elevator/ElevatorSystem.java:19]: Why only hardcode 1 elevator?

File: src/elevator/ElevatorSystem.java [hunk 4]
```diff
@@ -1,58 +1,77 @@
...
-					// TODO Auto-generated catch block
-					e.printStackTrace();
+		while (true) {
+			while (!inBoundRequests.isEmpty() || !outBoundRequests.isEmpty()) {
+				while (!inBoundRequests.isEmpty()) {
+					System.out.println("Elevator System: Sending messages to free elevator");
+					ele1.request(inBoundRequests.poll());
 				}
+				
+				synchronized (outBoundRequests) {
```
Comment (mpfan) [inline src/elevator/ElevatorSystem.java]: So you can't send messages to both elevators and scheduler at the same time. For now I think it's fine but I think  ideally we should have two threads.

File: src/elevator/ElevatorSystem.java (hunk 3 above)
Comment (dershao) [reply src/elevator/ElevatorSystem.java:19]: Just for iteration 1
It wouldn't be too big of a change for multiple elevators

**Pull request merged**